data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorTest.java

```diff
@@ -0,0 +1,151 @@
+    @Mock
+    private MaxMindServiceConfig maxMindServiceConfig;
+
+    @BeforeEach
+    void setUp() throws NoSuchFieldException, IllegalAccessException {
+
+        when(pluginSetting.getName()).thenReturn(PROCESSOR_PLUGIN_NAME);
+        when(pluginSetting.getPipelineName()).thenReturn(PROCESSOR_PIPELINE_NAME);
+        when(geoCodingProcessorConfig.getServiceType()).thenReturn(serviceTypeOptions);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService()).thenReturn(maxMindServiceConfig);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService().getLoadType())
+                .thenReturn(LoadTypeOptions.INMEMORY);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService().getCacheRefreshSchedule())
+                .thenReturn(Duration.ofSeconds(REFRESH_SCHEDULE));
+
+        DatabasePathURLConfig databasePathURLConfig1 = new DatabasePathURLConfig();
+        ReflectivelySetField.setField(DatabasePathURLConfig.class,
+                databasePathURLConfig1, "url", "./src/test/resources/mmdb-file/geo-lite2");
+        List<DatabasePathURLConfig> urlList = List.of(databasePathURLConfig1);
+        when(geoCodingProcessorConfig.getServiceType().getMaxMindService().getDatabasePath()).thenReturn(urlList);
+    }
+
+    @Test
+    void doExecuteTest() throws MalformedURLException, NoSuchFieldException, IllegalAccessException {
+        KeyConfig keyConfig = new KeyConfig();
+        final List<String> attributes = setAttributes();
+        ReflectivelySetField.setField(KeyConfig.class, keyConfig, "attributes", attributes);
+        ReflectivelySetField.setField(KeyConfig.class, keyConfig, "source", SOURCE);
+        ReflectivelySetField.setField(KeyConfig.class, keyConfig, "target", TARGET);
+
+        KeysConfig keysConfig1 = new KeysConfig();
+        ReflectivelySetField.setField(KeysConfig.class,
+                keysConfig1, "keyConfig", keyConfig);
+
+        List<KeysConfig> configs = new ArrayList<>();
+        configs.add(keysConfig1);
+        when(geoCodingProcessorConfig.getKeysConfig()).thenReturn(configs);
+        GeoIPProcessor geoIPProcessor = createObjectUnderTest();
+
+        Map<String, Object> geoData = new HashMap<>();
+        geoData.put("country_IsoCode", "US");
+        geoData.put("continent_name", "North America");
+        geoData.put("timezone", "America/Chicago");
+        geoData.put("country_name", "United States");
+
+        when(geoIPProcessorService.getGeoData(any(), any())).thenReturn(geoData);
+        ReflectivelySetField.setField(GeoIPProcessor.class, geoIPProcessor,
+                "geoIPProcessorService", geoIPProcessorService);
+        Collection<Record<Event>> records = geoIPProcessor.doExecute(setEventQueue());
+        for (final Record<Event> record : records) {
+            Event event = record.getData();
+            assertThat(event.get("/peer/ip", String.class), equalTo("136.226.242.205"));
```

Comment:
Need to test that events have been enriched with Geo data here.

Reply:
Already tested in GetGeoLite2DataTest file in two methods.
1.getGeoDataTest_without_attributes() 
2.getGeoDataTest_with_attributes().

Reply:
I'm looking for assertions like the following added here. This is to test that the event is populated with the prepared Geo data:
```
assertThat(event.get("location/country_name", String.class), equalTo("United States"));
assertThat(event.get("location/timezone", String.class), equalTo("America/Chicago"));
... ...
```

Reply:
This test case is to read the IP from the input JSON file based on the key present in YAML configuration. For example the path for reading the IP is peer/ip. Please find the below sample json input:

"peer" : {
"ip" : "1.2.3.4"
"host" : "example.org"
}
"status" : "success"

There is no enrichment in this test testcase to add assertions for country_name and timezone.
Enrichment related are tested in GetGeoLite2DataTest file in two methods.
1.getGeoDataTest_without_attributes()
2.getGeoDataTest_with_attributes()

Reply:
Thanks for addressing other comments! Looks pretty good other than this one: 

> There is no enrichment in this test testcase.

The test case is testing GeoIPProcessor's `doExecute` method, which enriches the events with the GeoIP data. And we want to verify that the events are actually enriched. This is a standard test for processors, [this one](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/mutate-event-processors/src/test/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorTests.java) is an example.

Something like this should work:
```suggestion
            assertThat(event.get("/peer/ip", String.class), equalTo("136.226.242.205"));
            assertThat(event.get("location/country_name", String.class), equalTo("United States"));
            assertThat(event.get("location/timezone", String.class), equalTo("America/Chicago"));
            assertThat(event.get("location/country_IsoCode", String.class), equalTo("US"));
            assertThat(event.get("location/continent_name", String.class), equalTo("North America"));
```

Reply:
I understand your argument, `doExecute()` indeed includes call to `getGeoData()`, which have been fully tested and are being mocked in this `doExecuteTest()`. But there's more than just `getGeoData()` in `doExecute()` which needs to be tested here, specifically this line that adds geo data to the event:
```
eventRecord.getData().put(key.getKeyConfig().getTarget(), geoData);
```
Imagine this line is missing, `doExecuteTest()` should catch it. But in its current form, it won't.

Reply:
Approving so we can move forward.